support/support.go

```diff
@@ -0,0 +1,227 @@
+	} else {
+		d.Config = dc
+	}
+
+	hr, err := client.GetDebugHealth(ctx)
+	if err != nil {
+		log.Error(ctx, "fetch health report", slog.Error(err))
+	} else {
+		d.HealthReport = &hr
+	}
+
+	exp, err := client.Experiments(ctx)
+	if err != nil {
+		log.Error(ctx, "fetch experiments", slog.Error(err))
+	} else {
+		d.Experiments = exp
+	}
+
+	return d
+}
+
+func NetworkInfo(ctx context.Context, client *codersdk.Client, log slog.Logger, agentID uuid.UUID) Network {
+	var n Network
+
+	// Get /api/v2/debug/coordinator
+	coordResp, err := client.Request(ctx, http.MethodGet, "/api/v2/debug/coordinator", nil)
+	if err != nil {
+		log.Error(ctx, "fetch coordinator debug page", slog.Error(err))
+	} else {
+		defer coordResp.Body.Close()
+		bs, err := io.ReadAll(coordResp.Body)
+		if err != nil {
+			log.Error(ctx, "read coordinator debug page", slog.Error(err))
+		} else {
+			n.CoordinatorDebug = string(bs)
+		}
+	}
+
+	// Get /api/v2/debug/tailnet
+	tailResp, err := client.Request(ctx, http.MethodGet, "/api/v2/debug/tailnet", nil)
+	if err != nil {
+		log.Error(ctx, "fetch tailnet debug page", slog.Error(err))
+	} else {
+		defer tailResp.Body.Close()
+		bs, err := io.ReadAll(tailResp.Body)
+		if err != nil {
+			log.Error(ctx, "read tailnet debug page", slog.Error(err))
+		} else {
+			n.TailnetDebug = string(bs)
+		}
+	}
+
+	if agentID != uuid.Nil {
+		connInfo, err := client.WorkspaceAgentConnectionInfo(ctx, agentID)
+		if err != nil {
+			log.Error(ctx, "fetch agent conn info", slog.Error(err), slog.F("agent_id", agentID.String()))
+		} else {
+			n.NetcheckLocal = &connInfo
+		}
+	} else {
+		log.Warn(ctx, "agent id required for agent connection info")
+	}
+
+	return n
+}
+
+func WorkspaceInfo(ctx context.Context, client *codersdk.Client, log slog.Logger, workspaceID, agentID uuid.UUID) Workspace {
+	var w Workspace
+
+	if workspaceID == uuid.Nil {
+		log.Error(ctx, "no workspace id specified")
+		return w
+	}
+
+	if agentID == uuid.Nil {
+		log.Error(ctx, "no agent id specified")
+	}
+
+	ws, err := client.Workspace(ctx, workspaceID)
+	if err != nil {
+		log.Error(ctx, "fetch workspace", slog.Error(err), slog.F("workspace_id", workspaceID))
+		return w
+	}
+
+	w.Workspace = ws
+
+	buildLogCh, closer, err := client.WorkspaceBuildLogsAfter(ctx, ws.LatestBuild.ID, 0)
+	if err != nil {
+		log.Error(ctx, "fetch provisioner job logs", slog.Error(err), slog.F("job_id", ws.LatestBuild.Job.ID.String()))
+	} else {
+		defer closer.Close()
+		for log := range buildLogCh {
+			w.BuildLogs = append(w.BuildLogs, log)
+		}
+	}
+
+	if len(w.Workspace.LatestBuild.Resources) == 0 {
+		log.Warn(ctx, "workspace build has no resources")
+		return w
+	}
+
+	agentLogCh, closer, err := client.WorkspaceAgentLogsAfter(ctx, agentID, 0, false)
+	if err != nil {
+		log.Error(ctx, "fetch agent startup logs", slog.Error(err), slog.F("agent_id", agentID.String()))
+	} else {
+		defer closer.Close()
+		for logChunk := range agentLogCh {
```

Comment:
Same as with workspace build, agent logs are limited to 1MB currently, but that assumption could change.